app/js/fm-player/controllers/PlayerCtrl.js

```diff
@@ -57,6 +87,76 @@ angular.module("sn.fm.player").controller("PlayerCtrl", [
             PlayerQueueResource.save({ uri: track.uri });
         };
 
+        /**
+         * Update `playlist` and `current` with queue data and currently playing track from the API
+         * @method refreshPlaylist
+         */
+        $scope.refreshPlaylist = function refreshPlaylistQueue(){
+            $q.all([
+                PlayerQueueResource.get().$promise,
+                PlayerTransportResource.get().$promise
+            ]).then(function(response){
+                $scope.playlist = response[0];
+                $scope.current = response[1];
+                $scope.playlist.unshift($scope.current);
+            });
+        };
+
+        /**
+         * Add currently playing track to playlist
+         * @method init
+         */
+        $scope.init = function init() {
+            $scope.playlist.unshift($scope.current);
+        };
+
+        /**
+         * On play event, remove previous track from playlist and set
+         * playback state variables
+         * @listens fm:player:play
+         */
+        $scope.$on("fm:player:play", function (event, data) {
+
+            if ($scope.playlist[1].spotify_uri === data.uri) { // jshint ignore:line
+                $scope.playlist.splice(0, 1);
+                $scope.paused = false;
+                $scope.current = $scope.playlist[0];
+            } else {
+                $scope.refreshPlaylist();
+                $scope.paused = false;
+            }
+
+        });
+
+        /**
+         * On pause event, update paused state
+         * @listens fm:player:pause
+         */
+        $scope.$on("fm:player:pause", function () {
+            $scope.paused = true;
+        });
+
+        /**
+         * On resume event, update paused state
+         * @listens fm:player:resume
+         */
+        $scope.$on("fm:player:resume", function () {
+            $scope.paused = false;
+        });
+
+        /**
+         * On add event, get track data and push to playlist
+         * @listens fm:player:pause
+         */
+        $scope.$on("fm:player:add", function (event, data) {
+            TracksResource.get({ id: data.uri }).$promise
+                .then(function(track){
+                    $scope.playlist.push(track);
+                });
+        });
```

Comment:
@jamesjwarren you should attach these event handlers in the scope rather then have them as anonymous functions